libs/core/async_sycl/tests/unit/sycl_vector_add_get_future.cpp

```diff
@@ -0,0 +1,205 @@
+//  Copyright (c) 2022 Gregor Daiß
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <algorithm>
+#include <exception>
+#include <iostream> 
+#include <string> 
+#include <vector>
+
+#include <hpx/futures/future.hpp>
+#include <hpx/hpx_init.hpp> 
+#include <hpx/local/future.hpp>
+#if defined(HPX_HAVE_SYCL)
+#include <hpx/async_sycl/sycl_future.hpp> 
+
+#include "common/sycl_vector_add_test_utils.hpp"
+
+#include <CL/sycl.hpp> 
+
+// Check compiler compatibility:
+// Needs to be done AFTER sycl include for HipSYCL
+// (intel dpcpp would be fine without the include)
+//
+// Will raise compile-time errors in case of unexpected behaviour! (Hence part of the unit test)
+// Uncomment the pragma message commands for more information about the compile passes 
+// (2 passes for hipsycl, 3 for dpcpp)
+#if defined(SYCL_LANGUAGE_VERSION)
+#if !defined(__INTEL_LLVM_COMPILER) && !defined(__HIPSYCL__)
+#warning "HPX-SYCL integration only tested with Intel oneapi and HipSYCL. \
+Utilized compiler appears to be neither of those!"
+#endif
+#else
+#error "Compiler does not seem to support SYCL! SYCL_LANGUAGE_VERSION is undefined!"
+#endif
```

Comment:
Should this be part of the library (instead of being part of the test)? Users might not run the test and still use the code in their own applications.

Reply:
Hmm, good point!

I briefly considered putting this into compiler_specific.hpp, however, even when building HPX for SYCL not necessarily all files are build with a SYCL compiler (for instance, with HIPSYCL we still build most things with clang++).

That being said, I think sycl_event_callback.hpp is a good place to put it. This file will be included by sycl_future.hpp and sycl_executor.hpp and is thus guaranteed to be included when using SYCL (with HPX) somewhere. I move the check to this header in c4df54708e5e4560e19b3ebdea390b45a9fe05de